frontend/src/app/core/browser/device.service.ts

```diff
@@ -1,10 +1,17 @@
 import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
 
 @Injectable({ providedIn: 'root' })
 export class DeviceService {
-  public mobileWidthTreshold = 680;
-
   public get isMobile():boolean {
-    return (window.innerWidth < this.mobileWidthTreshold);
+    return window.matchMedia('(max-width: 679px)').matches;
+  }
+
+  constructor() {
+    window.addEventListener('resize', () => {
```

Comment:
A resize listener is very expensive and it is basically registered on every page  because of the `MainMenuToggle`. Why do you need an Observable anyway? We usually only apply js-changes for mobile on page load and not on screen size..

Reply:
Yes You are right we actually don't need an observable in this case only on page loading. I'll undo these changes.